floodsub.go

```diff
@@ -171,7 +199,27 @@ func (p *PubSub) processLoop(ctx context.Context) {
 				continue
 			}
 		case msg := <-p.publish:
-			p.maybePublishMessage(p.host.ID(), msg.Message)
+			subs := p.getSubscriptions(msg) // call before goroutine!
+
+			select {
+			case p.throttleValidate <- struct{}{}:
+				go func(msg *Message) {
+					defer func() { <-p.throttleValidate }()
+
+					if p.validate(subs, msg) {
+						p.sendMsg <- sendReq{
+							from: p.host.ID(),
+							msg:  msg,
+						}
+
+					}
+				}(msg)
+			default:
+				log.Warning("could not acquire validator; dropping message")
```

Comment:
Perhaps we should log some more information about the message? The topic and the message id come to mind.